interfaces/core.go

```diff
@@ -35,10 +35,10 @@ type Plug struct {
 
 // Ref returns reference to a plug
 func (plug *Plug) Ref() PlugRef {
-	return plugRef(plug.PlugInfo)
+	return NewPlugRef(plug.PlugInfo)
 }
 
-func plugRef(plugInfo *snap.PlugInfo) PlugRef {
+func NewPlugRef(plugInfo *snap.PlugInfo) PlugRef {
```

Comment:
maybe I'm wrong but as far as I can tell we use this outside of repo only:  to get a string  or make a ConnRef,

so maybe what we want is really methods to get string reprs out of PlugInfo/SlotInfo,  and then some kind NewConnRef(slotInfo, plugInfo) instead?

Reply:
Good idea. Done.